ipatests/test_integration/test_trust.py

```diff
@@ -177,6 +161,108 @@ def test_upn_user_authentication_in_nonposix_trust(self):
                                 stdin_text=self.upn_password)
 
+    def test_filter_users(self):
+        """filter_users option is applied if SSSD starts offline.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/3983
+
+        When domain resolution order changed to prefer the AD domain
+        calling id root then No LDAP lookups for 'root@ad.domain'.
+        """
+        log_file = '{0}/sssd_nss.log'.format(paths.VAR_LOG_SSSD_DIR)
+        logsize = tasks.get_logsize(self.master, log_file)
+        tasks.backup_file(self.master, paths.SSSD_CONF)
+        try:
+            tasks.kinit_admin(self.master)
+            domains = '{0}:{1}'.format(self.ad_domain, self.master.domain.name)
+            self.master.run_command(['ipa', 'config-mod',
+                                     '--domain-resolution-order={0}'
+                                    .format(domains)])
+            # starting sssd in offline mode
+            self.master.run_command(['ipactl', 'stop'])
+            tasks.clear_sssd_cache(self.master)
+
+            self.master.run_command(['id', 'root'])
+            exp_txt1 = ('Looking up [root@{0}] in data provider'
+                        .format(self.ad_domain))
+            sssd_logs = self.master.get_file_contents(log_file)[logsize:]
+            assert exp_txt1.encode() not in sssd_logs
+
+        finally:
+            self.master.run_command(['ipactl', 'start'])
+            tasks.restore_files(self.master)
+            tasks.clear_sssd_cache(self.master)
+            self.master.run_command(['ipa', 'config-mod',
+                                     '--domain-resolution-order='])
+
+    @pytest.mark.xfail(
+        osinfo.id == 'fedora' and osinfo.version_number <= (30,),
+        reason='sssd ticket 3978', strict=True)
+    def test_filter_users_known_domains(self):
+        """Negative cache should use values from 'filter_users' config option.
+
+        Regression test for https://pagure.io/SSSD/sssd/issue/3978
+
+        When there are users in filter_users for nss provider then no
+        look up should be in data provider and added in Negative cache..
+        """
+        try:
+            tasks.backup_file(self.master, paths.SSSD_CONF)
+            log_file = '{0}/sssd_nss.log'.format(paths.VAR_LOG_SSSD_DIR)
+            logsize = tasks.get_logsize(self.master, log_file)
+            tasks.kinit_admin(self.master)
+            domains = '{0}:{1}'.format(self.ad_domain, self.master.domain.name)
+            self.master.run_command(['ipa', 'config-mod',
+                                     '--domain-resolution-order={0}'
+                                    .format(domains)])
+            user1 = 'some_user@some.domain'
+            aduser = 'testuser@{0}'.format(self.ad_domain)
+            domain = self.master.domain
+            tasks.modify_sssd_conf(
+                self.master, domain.name,
+                {
+                    'filter_users': '{0},{1}'.format(user1, aduser)
```

Comment:
according to `man sssd.conf` the `filter_users` option should be in [nss] section, not in the domain one

Reply:
I was wrong, it can be used in domain section.

Reply:
Adding in  [nss] is broader, [domain/...] would only filter for the given domain